fetchai/ledger/serialisation/transaction.py

```diff
@@ -33,7 +33,10 @@ def _map_contract_mode(payload: Transaction):
             return CHAIN_CODE
         assert payload.contract_digest is not None
 
-        return SMART_CONTRACT if payload.contract_address else SYNERGETIC
+        if payload.action == 'data':
```

Comment:
I don't think this is going to be enough here. We will probably need to introduce a specific piece of metadata into the transaction object itself to map the contract type (since there is on easily detectable way to determine the kind from the outside)

Reply:
Agreed, but let's do that as a separate submission, as it would demand backend changes